lib/diagnostic.rb

```diff
@@ -5,7 +5,7 @@ class Diagnostic
   # enumerable supplied.  Assume each entry implements a length
   # method that returns a Fixnum.
   def longest(enum)
-    # Your code goes here
+    enum.max_by {|x| x.length}
```

Comment:
Yet another solution would be to use `.max` :
`enum.max { |a, b| a.length <=> b.length }`